feat: Add AssociatedResources field to Manifest Spec (#1656)
kyma-project/lifecycle-manager: docs/technical-reference/api/manifest-cr.md

The resource is the default data that should be initialized for the module and is directly copied from **.spec.data** of the ModuleTemplate CR after normalizing it with the **namespace** for the synchronized module.

### **.spec.managedResources**

> The managedResources are the resources which are managed manually by the user. They are represented by the `group/version/kind` format. 

Reviewer: ```suggestion
The managed resources are the resources that are handled manually by the user. They are represented by the `group/version/kind` format. 
```

> The associated resources are the resources that are not managed by the lifecycle manager. They are represented by the `group/version/kind` format.  

Reviewer: ```suggestion
Associated resources are resources that are not managed by Lifecycle Manager. They are represented by the `group/version/kind` format.  
```

> If the module is managed and gets deleted from the Kyma CR, module deletion is suspended until all custom resources with GVK listed in the spec.managedResources are removed manually by the user.   

Reviewer: ```suggestion
Suppose the module is managed and gets deleted from the Kyma CR. In that case, the module deletion is suspended until all custom resources with GVK listed in the **spec.managedResources** are removed manually by the user.
```

> If the module is managed and gets deleted from the Kyma CR, module resources remain undeleted until all custom resources with GVK listed in the spec.managedResources are removed manually by the user.   

Reviewer 2: ```suggestion
If the module is managed and gets deleted from the Kyma CR, module deletion is suspended until all custom resources with GVK listed in the spec.managedResources are removed manually by the user.   
```

> Suppose the module is managed and gets deleted from the Kyma CR. In that case, the module deletion is suspended until all custom resources with GVK listed in the **spec.managedResources** are removed manually by the user.   

Reviewer: ```suggestion
Suppose the module is managed and gets deleted from the Kyma CR. In that case, the module deletion is suspended until all custom resources with GVK listed in the **spec.associatedResources** are removed manually by the user.   
```
